src/golang/lib/models/shared/types.go

```diff
@@ -0,0 +1,25 @@
+package shared
+
+type Type string
```

Comment:
Can we rename this to `ArtifactType` since we may have other types too, such as `OperatorType`. And can we also rename the file to something like `artifact.go` or `artifact_types.go` to indicate this is for Artifacts.